### PR TITLE
Chore: Make logger mutable for regression test

### PR DESCRIPTION
**Pull request by ying-jeanne**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: conf/defaults.ini [hunk 1]
```diff
@@ -671,6 +671,9 @@ content_types = text/html
 
 #################################### Logging ##########################
 [log]
+# Log is enabled by default, when disabled no log messages will be written
+enabled = true
```
Comment (ying-jeanne) [inline conf/defaults.ini:675]: not sure about make it public tho

File: pkg/tests/testinfra/testinfra.go [hunk 2]
```diff
@@ -288,6 +290,12 @@ func CreateGrafDir(t *testing.T, opts ...GrafanaOpts) (string, string) {
 			_, err = unifiedAlertingSection.NewKey("disabled_orgs", disableOrgStr)
 			require.NoError(t, err)
 		}
+		if !o.EnableLog {
+			logSection, err := cfg.NewSection("log")
```
Comment (papagian) [inline pkg/tests/testinfra/testinfra.go:294]: the `log` section is created [above](https://github.com/grafana/grafana/blob/649cd4bf7f8fa942e20bc0f2f1eeae7ffa5677db/pkg/tests/testinfra/testinfra.go#L180) so you can use `getOrCreateSection()` instead.

**Review by papagian — APPROVED** [review]
I have tested by enabling/disabling logging in the server and in a test and it works as expected 🎉 .
I'm not sure if we want to document this configuration thought.

**Review by yuri-tceretian — COMMENTED** [review]
LGMT. Ideally, I would prefer to log into the console in tests, or be able to configure it.

File: conf/defaults.ini (hunk 1 above)
Comment (yuri-tceretian) [reply conf/defaults.ini:675]: Let's document it.

**Pull request merged**